pkg/kvstore/etcd.go

```diff
@@ -439,6 +434,47 @@ func (e *EtcdClient) DeleteTree(path string) error {
 	return err
 }
 
+// StartWatch starts watching for changes in a prefix
+func (e *EtcdClient) StartWatch(w *Watcher) {
+	go func(w *Watcher) {
+		for {
+		recreateWatcher:
+			etcdWatch := e.cli.Watch(ctx.Background(), w.prefix, client.WithPrefix())
```

Comment:
You are re-creating the watcher for every watcher response.
This won't work for long.
You must keep the watcher and loop on reading from the channel as long as r.Err() is nil.
etcd is different from ZooKeeper...